`quarkus-maven-plugin:create` fails with missing codestart even for minimal project (no extensions)

### Describe the bug

Using the `quarkus-maven-plugin` version **3.22.1**, the `create` goal fails when generating a **minimal Quarkus project** with **no extensions explicitly requested**. The error is caused by a missing codestart artifact:

```
com.github.mcollovati:quarkus-hilla:jar:codestarts:24.8-SNAPSHOT
```

This happens even in a clean environment, and the extension in question is not part of the platform or requested by the user.

**Key Points**:

* No extensions are specified.
* The failure references a `SNAPSHOT` codestart from an external project (`quarkus-hilla`) that is not part of the platform BOM.
* The user has no control over this dependency being pulled.

### Expected behavior

* The CLI should never include codestarts for non-platform extensions unless explicitly requested.
* If codestarts are unavailable or broken, the plugin should **fail gracefully**, optionally with a warning or suggestion to use `-DnoCodestarts`.
* Project scaffolding should **succeed by default**, even if external codestarts are unavailable.

### Actual behavior

```
[ERROR] Failed to generate Quarkus project: Failed to resolve codestart artifact
com.github.mcollovati:quarkus-hilla:jar:codestarts:24.8-SNAPSHOT
(absent): Could not find artifact ...
```

### How to Reproduce?

```bash
mvn io.quarkus.platform:quarkus-maven-plugin:3.22.1:create \
  -DprojectGroupId=org.example \
  -DprojectArtifactId=my-app
```

### Output of `uname -a` or `ver`

Darwin ricferna-mac 24.4.0 Darwin Kernel Version 24.4.0: Fri Apr 11 18:33:47 PDT 2025; root:xnu-11417.101.15~117/RELEASE_ARM64_T6030 arm64

### Output of `java -version`

openjdk version "17.0.9" 2023-10-17 OpenJDK Runtime Environment GraalVM CE 17.0.9+9.1 (build 17.0.9+9-jvmci-23.0-b22) OpenJDK 64-Bit Server VM GraalVM CE 17.0.9+9.1 (build 17.0.9+9-jvmci-23.0-b22, mixed mode, sharing)

### Quarkus version or git rev

3.22

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven home: /opt/homebrew/Cellar/maven/3.9.9/libexec Java version: 17.0.9, vendor: GraalVM Community, runtime: /Users/ricferna/.sdkman/candidates/java/17.0.9-graalce Default locale: en_CA, platform encoding: UTF-8 OS name: "mac os x", version: "15.4.1", arch: "aarch64", family: "mac"

### Additional information

**Related Zulip Discussions**:

* [[Users thread](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/.60quarkus.20create.20cli.60.20is.20possibly.20broken.3F/near/517184095)](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/.60quarkus.20create.20cli.60.20is.20possibly.20broken.3F/near/517184095)
* [[Dev thread](https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Problem.20creating.20a.20new.20project.20with.20Quarkus.20Plugin/near/517185495)](https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Problem.20creating.20a.20new.20project.20with.20Quarkus.20Plugin/near/517185495)

Actually I can recreate something similar but for another artifact:

```
[DEBUG] Creating a new project with initial parameters: CreateCli{gav=TargetGAVGroup [groupId=org.acme, artifactId=code-with-quarkus, version=1.0.0-SNAPSHOT, gav=null], quarkusVersion=TargetQuarkusVersionGroup{stream=null, platformBom=null}, targetBuildTool=TargetBuildToolGroup [gradle=false, gradleKotlinDsl=false, jbang=false, maven=true], targetLanguage=TargetLanguageGroup [java=<<detect java runtime version>>, kotlin=false, scala=false, sourceType=null], codeGeneration=CodeGenerationGroup [includeCode=true, includeWrapper=true, includeDockerfiles=true, packageName=null], extensions=[], project=BaseCreateCommand [outputPath=null, registryClient=RegistryClientMixin [useRegistryClient=true], projectDirName=null, projectRootPath=null, targetDirectory=null, values={}], data={}, properties={}}
[DEBUG] Creating an app using the following settings: {project.artifact-id=code-with-quarkus, project.extensions=[picocli], codegen.no-buildtool-wrapper=false, data={}, codegen.no-dockerfiles=false, project.group-id=org.acme, project.app-config={}, codegen.no-code=false, project.version=1.0.0-SNAPSHOT, project.java-version=21}
[DEBUG] Resolving Quarkus extension catalog for TargetQuarkusVersionGroup{stream=null, platformBom=null}
[DEBUG] Resolved registry descriptor io.quarkus.registry:quarkus-registry-descriptor:json:1.0-SNAPSHOT from registry.quarkus.io
[DEBUG] Resolved registry descriptor com.redhat.quarkus.registry:quarkus-registry-descriptor:json:1.0-SNAPSHOT from registry.quarkus.redhat.com
[DEBUG] Non-platform extension catalogs were disabled for registry registry.quarkus.redhat.com
[DEBUG] Resolving platform catalog io.quarkus.registry:quarkus-platforms:json:1.0-SNAPSHOT
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-qpid-jms-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-cassandra-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-amazon-services-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-cxf-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-camel-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-debezium-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-blaze-persistence-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-google-cloud-services-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-operator-sdk-bom-quarkus-platform-descriptor:json:3.22.2:3.22.2
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-qpid-jms-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-cassandra-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-amazon-services-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-cxf-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-camel-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-debezium-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-blaze-persistence-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-google-cloud-services-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-operator-sdk-bom-quarkus-platform-descriptor:json:3.20.0:3.20.0
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-qpid-jms-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-cassandra-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-amazon-services-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-cxf-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-camel-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-debezium-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-blaze-persistence-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-google-cloud-services-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform extension catalog io.quarkus.platform:quarkus-operator-sdk-bom-quarkus-platform-descriptor:json:3.15.4:3.15.4
[DEBUG] Resolving platform catalog com.redhat.quarkus.registry:quarkus-platforms:json:1.0-SNAPSHOT
[DEBUG] Resolving platform extension catalog com.redhat.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:3.20.0.redhat-00002:3.20.0.redhat-00002
io.quarkus.registry.RegistryResolutionException: Failed to resolve extension catalog of com.redhat.quarkus.platform:quarkus-bom:pom:3.20.0.redhat-00002 from Maven repository central (https://repo.maven.apache.org/maven2)
	at io.quarkus.registry.client.maven.MavenPlatformExtensionsResolver.resolvePlatformExtensions(MavenPlatformExtensionsResolver.java:76)
	at io.quarkus.registry.client.RegistryClientDispatcher.resolvePlatformExtensions(RegistryClientDispatcher.java:44)
	at io.quarkus.registry.RegistryExtensionResolver.resolvePlatformExtensions(RegistryExtensionResolver.java:98)
	at io.quarkus.registry.ExtensionCatalogResolver.collectPlatformExtensions(ExtensionCatalogResolver.java:449)
	at io.quarkus.registry.ExtensionCatalogResolver.resolveExtensionCatalog(ExtensionCatalogResolver.java:397)
	at io.quarkus.cli.registry.RegistryClientMixin.getExtensionCatalog(RegistryClientMixin.java:114)
	at io.quarkus.cli.registry.RegistryClientMixin.createQuarkusProject(RegistryClientMixin.java:60)
	at io.quarkus.cli.create.BaseCreateCommand.build(BaseCreateCommand.java:223)
	at io.quarkus.cli.CreateCli.call(CreateCli.java:83)
	at io.quarkus.cli.CreateCli.call(CreateCli.java:22)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2045)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2465)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2457)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2419)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2277)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2421)
	at picocli.CommandLine.execute(CommandLine.java:2174)
	at io.quarkus.cli.QuarkusCli.run(QuarkusCli.java:117)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:143)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:77)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:48)
	at io.quarkus.cli.Main.main(Main.java:9)
Caused by: io.quarkus.bootstrap.resolver.maven.BootstrapMavenException: Failed to resolve artifact com.redhat.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:3.20.0.redhat-00002:3.20.0.redhat-00002
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveInternal(MavenArtifactResolver.java:169)
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolve(MavenArtifactResolver.java:154)
	at io.quarkus.registry.client.maven.MavenRegistryArtifactResolverWithCleanup.resolveAndCleanupOldTimestampedVersions(MavenRegistryArtifactResolverWithCleanup.java:68)
	at io.quarkus.registry.client.maven.MavenRegistryArtifactResolverWithCleanup.resolveArtifact(MavenRegistryArtifactResolverWithCleanup.java:37)
	at io.quarkus.registry.client.maven.MavenRegistryArtifactResolver.resolve(MavenRegistryArtifactResolver.java:14)
	at io.quarkus.registry.client.maven.MavenPlatformExtensionsResolver.resolvePlatformExtensions(MavenPlatformExtensionsResolver.java:49)
	... 22 more
Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: The following artifacts could not be resolved: com.redhat.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:3.20.0.redhat-00002:3.20.0.redhat-00002 (absent): Could not find artifact com.redhat.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:3.20.0.redhat-00002:3.20.0.redhat-00002 in central (https://repo.maven.apache.org/maven2)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:473)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifacts(DefaultArtifactResolver.java:261)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolveArtifact(DefaultArtifactResolver.java:243)
	at org.eclipse.aether.internal.impl.DefaultRepositorySystem.resolveArtifact(DefaultRepositorySystem.java:278)
	at io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver.resolveInternal(MavenArtifactResolver.java:164)
	... 27 more
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact com.redhat.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:json:3.20.0.redhat-00002:3.20.0.redhat-00002 in central (https://repo.maven.apache.org/maven2)
	at org.eclipse.aether.connector.basic.ArtifactTransportListener.transferFailed(ArtifactTransportListener.java:42)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector$TaskRunner.run(BasicRepositoryConnector.java:417)
	at org.eclipse.aether.connector.basic.BasicRepositoryConnector.get(BasicRepositoryConnector.java:260)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.performDownloads(DefaultArtifactResolver.java:537)
	at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:449)
	... 31 more

[ERROR] ❗  Unable to create project: Failed to resolve extension catalog of com.redhat.quarkus.platform:quarkus-bom:pom:3.20.0.redhat-00002 from Maven repository central (https://repo.maven.apache.org/maven2)
```

--- a/devtools/maven/src/main/java/io/quarkus/maven/CreateJBangMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/CreateJBangMojo.java
@@ -113,7 +113,7 @@ public class CreateJBangMojo extends AbstractMojo {
             throw new MojoExecutionException("Failed to resolve Quarkus extension catalog", e);
         }
 
-        final List<ResourceLoader> codestartsResourceLoader = codestartLoadersBuilder()
+        final List<ResourceLoader> codestartsResourceLoader = codestartLoadersBuilder(log)
                 .catalog(catalog)
                 .artifactResolver(mvn)
                 .build();

--- a/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/CreateProjectMojo.java
@@ -296,7 +296,7 @@ public class CreateProjectMojo extends AbstractMojo {
             catalog = CreateProjectHelper.completeCatalog(catalog, extensions, mvn);
             sanitizeOptions();
 
-            final List<ResourceLoader> codestartsResourceLoader = codestartLoadersBuilder()
+            final List<ResourceLoader> codestartsResourceLoader = codestartLoadersBuilder(log)
                     .catalog(catalog)
                     .artifactResolver(mvn)
                     .build();

--- a/devtools/maven/src/main/java/io/quarkus/maven/QuarkusProjectMojoBase.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/QuarkusProjectMojoBase.java
@@ -103,7 +103,7 @@ public abstract class QuarkusProjectMojoBase extends AbstractMojo {
         } else {
             final ExtensionCatalog extensionCatalog = resolveExtensionCatalog();
             final List<ResourceLoader> codestartsResourceLoader = CodestartResourceLoadersBuilder
-                    .codestartLoadersBuilder()
+                    .codestartLoadersBuilder(log)
                     .artifactResolver(artifactResolver())
                     .catalog(extensionCatalog)
                     .build();

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/extension/QuarkusExtensionCodestartCatalog.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/extension/QuarkusExtensionCodestartCatalog.java
@@ -12,6 +12,7 @@ import java.util.Map;
 import io.quarkus.devtools.codestarts.Codestart;
 import io.quarkus.devtools.codestarts.DataKey;
 import io.quarkus.devtools.codestarts.core.GenericCodestartCatalog;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 
 public final class QuarkusExtensionCodestartCatalog extends GenericCodestartCatalog<QuarkusExtensionCodestartProjectInput> {
 
@@ -78,9 +79,9 @@ public final class QuarkusExtensionCodestartCatalog extends GenericCodestartCata
         GIT
     }
 
-    public static QuarkusExtensionCodestartCatalog fromBaseCodestartsResources()
+    public static QuarkusExtensionCodestartCatalog fromBaseCodestartsResources(MessageWriter log)
             throws IOException {
-        final Map<String, Codestart> codestarts = loadCodestartsFromResources(getCodestartResourceLoaders(),
+        final Map<String, Codestart> codestarts = loadCodestartsFromResources(getCodestartResourceLoaders(log),
                 QUARKUS_EXTENSION_CODESTARTS_DIR);
         return new QuarkusExtensionCodestartCatalog(codestarts.values());
     }

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/jbang/QuarkusJBangCodestartCatalog.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/jbang/QuarkusJBangCodestartCatalog.java
@@ -12,6 +12,7 @@ import java.util.Map;
 import io.quarkus.devtools.codestarts.Codestart;
 import io.quarkus.devtools.codestarts.DataKey;
 import io.quarkus.devtools.codestarts.core.GenericCodestartCatalog;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.platform.descriptor.loader.json.ResourceLoader;
 
 public final class QuarkusJBangCodestartCatalog extends GenericCodestartCatalog<QuarkusJBangCodestartProjectInput> {
@@ -48,9 +49,9 @@ public final class QuarkusJBangCodestartCatalog extends GenericCodestartCatalog<
         super(codestarts);
     }
 
-    public static QuarkusJBangCodestartCatalog fromBaseCodestartsResources()
+    public static QuarkusJBangCodestartCatalog fromBaseCodestartsResources(MessageWriter log)
             throws IOException {
-        final Map<String, Codestart> codestarts = loadCodestartsFromResources(getCodestartResourceLoaders(),
+        final Map<String, Codestart> codestarts = loadCodestartsFromResources(getCodestartResourceLoaders(log),
                 QUARKUS_JBANG_CODESTARTS_DIR);
         return new QuarkusJBangCodestartCatalog(codestarts.values());
     }

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartCatalog.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/codestarts/quarkus/QuarkusCodestartCatalog.java
@@ -10,7 +10,6 @@ import java.io.IOException;
 import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
@@ -28,6 +27,7 @@ import io.quarkus.devtools.codestarts.CodestartStructureException;
 import io.quarkus.devtools.codestarts.CodestartType;
 import io.quarkus.devtools.codestarts.DataKey;
 import io.quarkus.devtools.codestarts.core.GenericCodestartCatalog;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.project.extensions.Extensions;
 import io.quarkus.maven.dependency.ArtifactCoords;
 import io.quarkus.platform.catalog.processor.ExtensionProcessor;
@@ -86,22 +86,23 @@ public final class QuarkusCodestartCatalog extends GenericCodestartCatalog<Quark
         this.extensionsMapping = extensionsMapping;
     }
 
-    public static QuarkusCodestartCatalog fromBaseCodestartsResources(Map<String, Extension> extensionsMapping)
+    public static QuarkusCodestartCatalog fromBaseCodestartsResources(MessageWriter log,
+            Map<String, Extension> extensionsMapping)
             throws IOException {
-        final Map<String, Codestart> codestarts = loadCodestartsFromResources(getCodestartResourceLoaders(),
+        final Map<String, Codestart> codestarts = loadCodestartsFromResources(getCodestartResourceLoaders(log),
                 QUARKUS_CODESTARTS_DIR);
         return new QuarkusCodestartCatalog(codestarts.values(), extensionsMapping);
     }
 
-    public static QuarkusCodestartCatalog fromBaseCodestartsResources()
+    public static QuarkusCodestartCatalog fromBaseCodestartsResources(MessageWriter log)
             throws IOException {
-        return fromBaseCodestartsResources(Collections.emptyMap());
+        return fromBaseCodestartsResources(log, Map.of());
     }
 
     public static QuarkusCodestartCatalog fromExtensionsCatalogAndDirectories(
-            ExtensionCatalog catalog, Collection<Path> directories)
+            MessageWriter log, ExtensionCatalog catalog, Collection<Path> directories)
             throws IOException {
-        final List<ResourceLoader> loaders = getCodestartResourceLoaders(catalog);
+        final List<ResourceLoader> loaders = getCodestartResourceLoaders(log, catalog);
         final Map<String, Codestart> codestarts = loadCodestartsFromResources(loaders,
                 QUARKUS_CODESTARTS_DIR);
         for (Path directory : directories) {

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateExtensionCommandHandler.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateExtensionCommandHandler.java
@@ -63,7 +63,7 @@ public class CreateExtensionCommandHandler {
             throws QuarkusCommandException {
         try {
             final QuarkusExtensionCodestartCatalog catalog = QuarkusExtensionCodestartCatalog
-                    .fromBaseCodestartsResources();
+                    .fromBaseCodestartsResources(log);
             catalog.createProject(input).generate(newExtensionDir);
 
             final String extensionDirName = newExtensionDir.getFileName().toString();

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/CodestartResourceLoadersBuilder.java
@@ -19,6 +19,7 @@ import org.eclipse.aether.artifact.Artifact;
 
 import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
 import io.quarkus.bootstrap.util.DependencyUtils;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.platform.descriptor.loader.json.ResourceLoader;
 import io.quarkus.registry.catalog.Extension;
 import io.quarkus.registry.catalog.ExtensionCatalog;
@@ -33,6 +34,7 @@ public final class CodestartResourceLoadersBuilder {
     private MavenArtifactResolver artifactResolver;
     private String baseCodestartsArtifactCoords = BASE_CODESTARTS_ARTIFACT_COORDS;
     private Collection<String> extraCodestartsArtifactCoords = new ArrayList<>();
+    private MessageWriter log;
 
     private static String retrieveBaseCodestartsArtifactCoords() {
         final String artifact = PropertiesUtil.getProperty(BASE_CODESTARTS_ARTIFACT_PROPERTY);
@@ -52,19 +54,20 @@ public final class CodestartResourceLoadersBuilder {
         }
     }
 
-    private CodestartResourceLoadersBuilder() {
+    private CodestartResourceLoadersBuilder(MessageWriter log) {
+        this.log = log;
     }
 
-    public static CodestartResourceLoadersBuilder codestartLoadersBuilder() {
-        return new CodestartResourceLoadersBuilder();
+    public static CodestartResourceLoadersBuilder codestartLoadersBuilder(MessageWriter log) {
+        return new CodestartResourceLoadersBuilder(log);
     }
 
-    public static List<ResourceLoader> getCodestartResourceLoaders() {
-        return codestartLoadersBuilder().build();
+    public static List<ResourceLoader> getCodestartResourceLoaders(MessageWriter log) {
+        return codestartLoadersBuilder(log).build();
     }
 
-    public static List<ResourceLoader> getCodestartResourceLoaders(ExtensionCatalog catalog) {
-        return codestartLoadersBuilder().catalog(catalog).build();
+    public static List<ResourceLoader> getCodestartResourceLoaders(MessageWriter log, ExtensionCatalog catalog) {
+        return codestartLoadersBuilder(log).catalog(catalog).build();
     }
 
     public CodestartResourceLoadersBuilder catalog(ExtensionCatalog catalog) {
@@ -94,13 +97,14 @@ public final class CodestartResourceLoadersBuilder {
 
     public List<ResourceLoader> build() {
         return getCodestartResourceLoaders(baseCodestartsArtifactCoords, extraCodestartsArtifactCoords, catalog,
-                artifactResolver == null ? QuarkusProjectHelper.artifactResolver() : artifactResolver);
+                artifactResolver == null ? QuarkusProjectHelper.artifactResolver() : artifactResolver, log);
     }
 
     private static List<ResourceLoader> getCodestartResourceLoaders(String baseCodestartsArtifactCoords,
             Collection<String> extraCodestartsArtifactCoords,
             ExtensionCatalog catalog,
-            MavenArtifactResolver mvn) {
+            MavenArtifactResolver mavenArtifactResolver,
+            MessageWriter log) {
 
         final Map<String, Artifact> codestartsArtifacts = new LinkedHashMap<>();
 
@@ -146,10 +150,11 @@ public final class CodestartResourceLoadersBuilder {
         final List<ResourceLoader> codestartResourceLoaders = new ArrayList<>(codestartsArtifacts.size());
         for (Artifact a : codestartsArtifacts.values()) {
             try {
-                final File artifactFile = mvn.resolve(a).getArtifact().getFile();
+                final File artifactFile = mavenArtifactResolver.resolve(a).getArtifact().getFile();
                 codestartResourceLoaders.add(resolveFileResourceLoader(artifactFile));
             } catch (Exception e) {
-                throw new RuntimeException("Failed to resolve codestart artifact " + a, e);
+                log.warn("Unable to resolve codestart artifact for %s: %s", a, e.getMessage());
+                continue;
             }
         }
         return codestartResourceLoaders;

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/QuarkusProjectHelper.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/QuarkusProjectHelper.java
@@ -146,7 +146,7 @@ public class QuarkusProjectHelper {
     public static QuarkusProject getProject(Path projectDir, ExtensionCatalog catalog, BuildTool buildTool,
             JavaVersion javaVersion,
             MessageWriter log) {
-        return QuarkusProject.of(projectDir, catalog, getCodestartResourceLoaders(catalog),
+        return QuarkusProject.of(projectDir, catalog, getCodestartResourceLoaders(log, catalog),
                 log, buildTool, javaVersion);
     }
 
@@ -161,7 +161,7 @@ public class QuarkusProjectHelper {
     public static QuarkusProject getProject(Path projectDir, ExtensionCatalog catalog, ExtensionManager extManager,
             JavaVersion javaVersion,
             MessageWriter log) {
-        return QuarkusProject.of(projectDir, catalog, getCodestartResourceLoaders(catalog),
+        return QuarkusProject.of(projectDir, catalog, getCodestartResourceLoaders(log, catalog),
                 log, extManager, javaVersion);
     }
 

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenProjectBuildFile.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/buildfile/MavenProjectBuildFile.java
@@ -121,7 +121,7 @@ public class MavenProjectBuildFile extends BuildFile {
         }
         final MavenProjectBuildFile extensionManager = new MavenProjectBuildFile(projectDir, extensionCatalog,
                 projectModel, deps, managedDeps, projectProps, projectPom == null ? null : artifactResolver);
-        final List<ResourceLoader> codestartResourceLoaders = codestartLoadersBuilder().catalog(extensionCatalog)
+        final List<ResourceLoader> codestartResourceLoaders = codestartLoadersBuilder(log).catalog(extensionCatalog)
                 .artifactResolver(artifactResolver).build();
         final JavaVersion javaVersion = resolveJavaVersion(projectProps);
         return QuarkusProject.of(projectDir, extensionCatalog,

--- a/independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/FakeExtensionCatalog.java
+++ b/independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/FakeExtensionCatalog.java
@@ -6,6 +6,7 @@ import java.io.UncheckedIOException;
 import java.util.Map;
 
 import io.quarkus.devtools.codestarts.quarkus.QuarkusCodestartCatalog;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.registry.catalog.ExtensionCatalog;
 
 public final class FakeExtensionCatalog {
@@ -20,6 +21,7 @@ public final class FakeExtensionCatalog {
     private static QuarkusCodestartCatalog getQuarkusCodestartCatalog() {
         try {
             return QuarkusCodestartCatalog.fromBaseCodestartsResources(
+                    MessageWriter.info(),
                     QuarkusCodestartCatalog.buildExtensionsMapping(FAKE_EXTENSION_CATALOG.getExtensions()));
         } catch (IOException e) {
             throw new UncheckedIOException(e);

--- a/independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/PlatformAwareTestBase.java
+++ b/independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/PlatformAwareTestBase.java
@@ -7,6 +7,7 @@ import java.util.Properties;
 
 import org.junit.jupiter.api.extension.RegisterExtension;
 
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.project.QuarkusProjectHelper;
 import io.quarkus.platform.descriptor.loader.json.ResourceLoader;
 import io.quarkus.platform.tools.ToolsUtils;
@@ -21,7 +22,7 @@ public class PlatformAwareTestBase {
     private Properties quarkusProps;
 
     protected List<ResourceLoader> getCodestartsResourceLoaders() {
-        return getCodestartResourceLoaders(getExtensionsCatalog());
+        return getCodestartResourceLoaders(MessageWriter.info(), getExtensionsCatalog());
     }
 
     protected ExtensionCatalog getExtensionsCatalog() {

--- a/independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/codestarts/QuarkusCodestartTest.java
+++ b/independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/codestarts/QuarkusCodestartTest.java
@@ -39,6 +39,7 @@ import org.junit.jupiter.api.extension.ExtensionContext;
 import io.quarkus.devtools.codestarts.quarkus.QuarkusCodestartCatalog;
 import io.quarkus.devtools.codestarts.quarkus.QuarkusCodestartCatalog.Language;
 import io.quarkus.devtools.codestarts.quarkus.QuarkusCodestartProjectInput;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.project.BuildTool;
 import io.quarkus.devtools.project.QuarkusProjectHelper;
 import io.quarkus.devtools.testing.SnapshotTesting;
@@ -354,7 +355,7 @@ public class QuarkusCodestartTest implements BeforeAllCallback, AfterAllCallback
     }
 
     protected List<ResourceLoader> getCodestartsResourceLoaders() {
-        return codestartLoadersBuilder()
+        return codestartLoadersBuilder(MessageWriter.info())
                 .catalog(getExtensionsCatalog())
                 .addExtraCodestartsArtifactCoords(artifacts)
                 .build();

--- a/independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/codestarts/extension/QuarkusExtensionCodestartGenerationTest.java
+++ b/independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/codestarts/extension/QuarkusExtensionCodestartGenerationTest.java
@@ -11,6 +11,7 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInfo;
 
 import io.quarkus.devtools.codestarts.extension.QuarkusExtensionCodestartCatalog.QuarkusExtensionData;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.testing.SnapshotTesting;
 
 class QuarkusExtensionCodestartGenerationTest {
@@ -57,7 +58,7 @@ class QuarkusExtensionCodestartGenerationTest {
     }
 
     private QuarkusExtensionCodestartCatalog getCatalog() throws IOException {
-        return QuarkusExtensionCodestartCatalog.fromBaseCodestartsResources();
+        return QuarkusExtensionCodestartCatalog.fromBaseCodestartsResources(MessageWriter.info());
     }
 
 }

--- a/independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/codestarts/jbang/QuarkusJBangCodestartGenerationTest.java
+++ b/independent-projects/tools/devtools-testing/src/test/java/io/quarkus/devtools/codestarts/jbang/QuarkusJBangCodestartGenerationTest.java
@@ -16,6 +16,7 @@ import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInfo;
 
 import io.quarkus.devtools.codestarts.utils.NestedMaps;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.testing.SnapshotTesting;
 
 class QuarkusJBangCodestartGenerationTest {
@@ -61,7 +62,7 @@ class QuarkusJBangCodestartGenerationTest {
     }
 
     private QuarkusJBangCodestartCatalog getCatalog() throws IOException {
-        return QuarkusJBangCodestartCatalog.fromBaseCodestartsResources();
+        return QuarkusJBangCodestartCatalog.fromBaseCodestartsResources(MessageWriter.info());
     }
 
 }

--- a/integration-tests/devtools/src/test/java/io/quarkus/platform/catalog/CatalogCompatibilityTest.java
+++ b/integration-tests/devtools/src/test/java/io/quarkus/platform/catalog/CatalogCompatibilityTest.java
@@ -8,6 +8,7 @@ import java.util.List;
 import org.junit.jupiter.api.Test;
 
 import io.quarkus.devtools.codestarts.quarkus.QuarkusCodestartCatalog;
+import io.quarkus.devtools.messagewriter.MessageWriter;
 import io.quarkus.devtools.project.CodestartResourceLoadersBuilder;
 import io.quarkus.devtools.project.QuarkusProjectHelper;
 import io.quarkus.devtools.testing.PlatformAwareTestBase;
@@ -65,7 +66,7 @@ public class CatalogCompatibilityTest extends PlatformAwareTestBase {
 
     private static void checkCodestarts(ExtensionCatalog extensionCatalog, CatalogProcessor processed) throws IOException {
         final List<ResourceLoader> codestartResourceLoaders = CodestartResourceLoadersBuilder
-                .getCodestartResourceLoaders(extensionCatalog);
+                .getCodestartResourceLoaders(MessageWriter.info(), extensionCatalog);
         final QuarkusCodestartCatalog quarkusCodestartCatalog = QuarkusCodestartCatalog
                 .fromExtensionsCatalog(extensionCatalog, codestartResourceLoaders);
         assertThat(quarkusCodestartCatalog.getCodestarts()).isNotNull();

